### ISSUE TITLE
[Elixir] Implement new Concept Exercise: `io`

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/v3#2291]
## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

There already exist a concept exercise ([`try-rescue-else-after`](https://github.com/exercism/v3/tree/master/languages/elixir/exercises/concept/try-rescue-else-after)) that depends on the `io` concept. One of the goals of this exercise is to enable the student to solve the `try-rescue-else-after` exercise. The other goal being to teach interacting with the standard IO devices and files.

## Learning objectives

- the `IO` module, including `puts`, `warn`, `inspect` - writing to stdio and stderr
- maybe also reading from stdio?
- what are IO devices?
- the `File` module, including `open`, `close`, `write`, `read` - maybe compare `File.open` + `IO.write` vs `File.write`?

It might turn out that putting together `io` and `file` concepts will be too much, or it's impossible to come up with a story that makes sense for both. If that's the case, let's try to split this exercise into an `io` exercise that only requires `string` or `basics` so that it appears very early on and teaches simple debugging techniques (`IO.puts("we reached this line of code") 😉`), and a more advanced `file` exercise that will be the preparation for `try-rescue-else-after`.

## Out of scope

?

## Concepts

- `io`
- `file`

## Prerequisites

- `string`* (maybe, unless we can limit ourselves to _just_ string literals then `basics` is enough)
- `tuples` (`File.open` returns an ok/error-tuple)
- `atoms` (an IO device can be an atom)
- `pids` (an IO device can be a PID)

## Resources to refer to

- https://hexdocs.pm/elixir/IO.html
- https://hexdocs.pm/elixir/File.html
- https://elixir-lang.org/getting-started/io-and-the-file-system.html

## Story

When coming up with a story, keep in mind that the students do not see the output of their code when they use the web editor. They only see if the tests passed. Look through existing stories (`/reference/stories/`) to find inspiration or an existing story that can work for this exercise.

## Implementation

- Use [`CaptureIO`](https://hexdocs.pm/ex_unit/ExUnit.CaptureIO.html#content) in tests to check what was written to IO devices

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by angelikatyborska** [comment]
An exercise for an IO concept was implemented, so I'll close this. But we still need to figure out what to do about the requirements for https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/rpn-calculator-output/.meta/example.ex
